packages/flame/lib/src/components/text_box_component.dart

```diff
@@ -210,17 +216,18 @@ class TextBoxComponent<T extends TextRenderer> extends PositionComponent {
     _textRenderer.render(c, line, Vector2(_boxConfig.margins.left, dy));
   }
 
-  void redrawLater() async {
-    _cache = await _redrawCache();
+  Future<void> redraw() async {
+    final newSize = _recomputeSize();
+    _cache = await _fullRenderAsImage(newSize);
+    size = newSize;
   }
 
   @override
   void update(double dt) {
     super.update(dt);
     _lifeTime += dt;
     if (_previousChar != currentChar) {
-      size = _recomputeSize();
-      redrawLater();
+      redraw();
```

Comment:
I drawing was too expensive I considered that these futures could potentially race. however (1) testing several examples I never saw that happening, (2) seems very unlikely for reasonable typing speed vs rendering speeds, and (3) would be extremely convoluted to keep track of futures and have them cancelled. but if you think it's important I can address that on a followup.